### PR TITLE
Saving form information when user go to another modal

### PR DESCRIPTION
**Pull request by YaroslavOvdii**
@tonytlwu @squallstar 
## Issue
https://github.com/Fliplet/fliplet-studio/issues/3755

## Description
Created a save function witch is saving data on change icon or splash screen.

## Screenshots/screencasts
issue-3755

## Backward compatibility

This change is fully backward compatible.

## Notes
Slightly altered saving functions so that they return a promise.

### CODE REVIEW

File: js/interface.js [hunk 1]
```diff
@@ -2371,37 +2383,54 @@ $('[name="fl-store-keywords"]').on('tokenfield:createtoken', function (e) {
...
+    Fliplet.Studio.emit('overlay', {
+      name: 'app-settings',
+      options: {
+        size: 'large',
+        title: 'App Settings',
+        section: 'appSettingsGeneral',
+        appId: Fliplet.Env.get('appId')
+      }
+    });
+  })
```
Comment (tonytlwu) [inline js/interface.js]: Move `.catch()` to this line

```suggestion
  }).catch(function (err) {
```

File: js/interface.js [hunk 2]
```diff
@@ -2371,37 +2383,54 @@ $('[name="fl-store-keywords"]').on('tokenfield:createtoken', function (e) {
...
+        section: 'launchAssets',
+        appId: Fliplet.Env.get('appId')
+      }
+    });
+  })
+  .catch(function (err) {
+    Fliplet.Modal.alert({
+      message: Fliplet.parseError(err)
+    })
+  })
```
Comment (tonytlwu) [inline js/interface.js]:
```suggestion
  });
```

File: js/interface.js [hunk 3]
```diff
@@ -2371,37 +2383,54 @@ $('[name="fl-store-keywords"]').on('tokenfield:createtoken', function (e) {
...
+        section: 'appSettingsGeneral',
+        appId: Fliplet.Env.get('appId')
+      }
+    });
+  })
+  .catch(function (err) {
+    Fliplet.Modal.alert({
+      message: Fliplet.parseError(err)
+    })
+  })
```
Comment (tonytlwu) [inline js/interface.js]:
```suggestion
  });
```

Comment (YaroslavOvdii) [reply js/interface.js]: @tonytlwu 
Fixed

**Pull request merged**